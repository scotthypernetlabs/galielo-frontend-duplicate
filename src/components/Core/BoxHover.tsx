import { Button, Grid } from "@material-ui/core";
import React from "react";

interface StationBoxHoverProps {
  hover: boolean;
  onClickButton1: any;
  onClickButton2: any;
  textButton1: string;
  textButton2: string;
}

const BoxHover: React.SFC<StationBoxHoverProps> = (
  props: StationBoxHoverProps
) => {
  const { hover, onClickButton1, onClickButton2, textButton1, textButton2 } = props;
  let className = "station-hover-grid";
  if (!hover) {
    className += " hidden";
  }
  return (
    <Grid container className={className}>
      <Grid className="station-hover-button-container">
        <Button
          size="small"
          variant="contained"
          color="primary"
          onClick={onClickButton1}
        >
          {textButton1}
        </Button>
        <Button
          size="small"
          variant="contained"
          color="primary"
          onClick={onClickButton2}
        >
          {textButton2}
        </Button>
      </Grid>
    </Grid>
  );
};

export default BoxHover;
