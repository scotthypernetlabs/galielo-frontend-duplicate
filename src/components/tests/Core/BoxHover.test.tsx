import { Box, Button, Grid } from "@material-ui/core";
import { mount } from "enzyme";
import BoxHover from "../../Core/BoxHover";
import React from "react";

describe("BoxHover Component", () => {
  const componentProps = {
    hover: true,
    onClickButton1: jest.fn(),
    onClickButton2: jest.fn(),
    textButton1: "Button1 Text",
    textButton2: "Button2 Text"
  };

  const Wrapper = (props: any) => {
    return <BoxHover {...componentProps} />;
  };

  const boxHover = mount(<Wrapper />);

  componentProps.hover = false;

  const WrapperNoHover = (props: any) => {
    return <BoxHover {...componentProps} />;
  };

  const boxNoHover = mount(<WrapperNoHover />);

  it("buttons should have the right text", () => {
    expect(boxHover.find(Button)).toHaveLength(2);
    expect(
      boxHover
        .find(Button)
        .at(0)
        .childAt(0)
        .text()
    ).toBe(componentProps.textButton1);
    expect(
      boxHover
        .find(Button)
        .at(1)
        .childAt(0)
        .text()
    ).toBe(componentProps.textButton2);
  });

  it("first button should call onClickButton1", () => {
    boxHover
      .find(Button)
      .at(0)
      .simulate("click");
    expect(boxHover.childAt(0).props().onClickButton1).toHaveBeenCalled();
  });

  it("second button should call onClickButton2", () => {
    boxHover
      .find(Button)
      .at(1)
      .simulate("click");
    expect(boxHover.childAt(0).props().onClickButton2).toHaveBeenCalled();
  });

  it("no hover should have no buttons", () => {
    expect(
      boxNoHover
        .find(Box)
        .at(0)
        .props().display
    ).toBe("none");
  });

  it("should match snapshot", () => {
    expect(boxHover).toMatchSnapshot();
  });
});
