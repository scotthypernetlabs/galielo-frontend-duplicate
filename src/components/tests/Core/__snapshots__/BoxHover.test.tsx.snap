// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxHover Component should match snapshot 1`] = `ReactWrapper {}`;
